### PR TITLE
Convert shooter into a submodule

### LINKED ISSUE #383: Update shooter to latest version 0.6.1, targeting Minetest 5.0.0
`shooter` has gotten some major updates recently, including the use of Raycast for simplifying the code by leaps and bounds. So instead of just fixing the eye-height, I suggest we update `shooter` to v0.6.1, which is the version targeted at Minetest 5.0.0, IIRC.

### PR DESCRIPTION
**Pull request by ClobberXD**
(#398 resurrected -_-)

`mods/pvp/shooter` now points to https://github.com/MT-CTF/shooter.

This also updates `shooter` to v0.6.1, which includes major changes in every area, like switching to a raycast-based implementation, proper spread, real shotguns, etc.

Although not strictly required, I've replaced the now-deprecated gun names with the new ones. e.g.
`shooter:pistol` --> `shooter_guns:pistol_loaded`

Testing needed! Closes #383

### CODE REVIEW

**Comment by ClobberXD** [comment]
@MT-CTF/developers Rebasing PRs that modify submodules is a royal pain, and it'd be great if this PR can be tested and reviewed ASAP. To test, the tester just has to play around with guns, to make sure nothing is broken. Additionally, treasure-spawning could also be verified to be working properly, although I highly doubt there'd be problems here.

**Comment by ClobberXD** [comment]
I've simplified `update.sh` and the proposed implementation is (arguably) the right way to update submodules. The added benefit of the new script is that missing would automatically be initialised and updated to the target commit.

File: .gitmodules [hunk 1]
```diff
@@ -4,3 +4,6 @@
 [submodule "mods/ctf/ctf_map/maps"]
 	path = mods/ctf/ctf_map/maps
 	url = https://github.com/MT-CTF/maps.git
+[submodule "mods/pvp/shooter"]
+	path = mods/pvp/shooter
+	url = https://github.com/MT-CTF/shooter
```
Comment (Panquesito7) [inline .gitmodules]:
```suggestion
	url = https://github.com/MT-CTF/shooter.git
```

Comment (ClobberXD) [reply .gitmodules]: https://github.com/MT-CTF/capturetheflag/compare/341230e350a5ac0e0f1cf6c8495649ff4b4ae52b..f25ac0aadf028559afd65768bb050add2358a233#diff-8903239df476d7401cf9e76af0252622

**Comment by ClobberXD** [comment]
We're currently using the high priority label to suggest that a PR should be reviewed before the others, due to reasons like critical bug, blocker, etc. So I'm unmarking this PR as high-priority - this isn't blocking anything, and there are also some issues like the grenade kill being attributed to the victim themselves (!!!).

**Comment by ClobberXD** [comment]
`shooter` is subtly broken in ways multiple ways. Removing this from the 2.0.0 milestone; we can re-consider this for a 2.1 release...

**Pull request closed without merging**